spec/jobs/cypress_viewport_updater/google_analytics_reports_spec.rb

```diff
@@ -0,0 +1,147 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe CypressViewportUpdater::GoogleAnalyticsReports do
+  VCR.configure do |c|
+    # the following filter is used on requests to
+    # https://analyticsreporting.googleapis.com/v4/reports:batchGet
+    c.filter_sensitive_data('removed') do |interaction|
+      if interaction.request.headers['Authorization']
+        if (match = interaction.request.headers['Authorization'].first.match(/^Bearer.+/))
+          match[0]
+        end
+      end
+    end
+
+    c.filter_sensitive_data('{"access_token":"removed","expires_in":3599,"token_type":"Bearer"}') do |interaction|
+      if (match = interaction.response.body.match(/^{\"access_token.+/))
+        match[0]
+      end
+    end
+  end
+
+  before do
+    allow(Google::Auth::ServiceAccountCredentials).to receive(:make_creds).and_return(true)
+  end
+
+  describe '#new' do
+    it 'returns a new instance' do
+      google_analytics = CypressViewportUpdater::GoogleAnalyticsReports.new
+      expect(google_analytics).to be_an_instance_of(described_class)
+    end
+  end
+
+  describe 'before #request_report is called' do
```

Comment:
Really minor comment: this could be a `context` block instead of a `describe`.

Reply:
Thanks Darius. I updated those two lines.